src/tools/perf/lib/libperf.c

```diff
@@ -437,6 +437,32 @@ static ucs_status_t ucx_perf_test_check_params(ucx_perf_params_t *params)
     return UCS_OK;
 }
 
+void uct_perf_iface_ep_flush_b(ucx_perf_context_t *perf, int peer_index)
+{
+    uct_ep_h ep = perf->uct.peers[peer_index].ep;
+    uct_completion_t comp;
+    ucs_status_t status;
+    int started;
+
+    started    = 0;
+    comp.func  = NULL;
+    comp.count = 2;
+    do {
+        if (!started) {
+            status = uct_ep_flush(ep, 0, &comp);
+            if (status == UCS_OK) {
+                --comp.count;
+            } else if (status == UCS_INPROGRESS) {
+                started = 1;
+            } else if (status != UCS_ERR_NO_RESOURCE) {
+                ucs_error("uct_ep_flush() failed: %s", ucs_status_string(status));
+                return;
+            }
+        }
+        uct_worker_progress(perf->uct.worker);
+    } while (comp.count > 1);
```

Comment:
I think, it is more clear to have `(comp.count != 0)`, wdyt?

Reply:
remove it , this is part of another future fix:) but anyway if it would reach 0 would also need to specify valid callback function, and i wanted to avoid it

Reply:
Ah, I see
thank you